homeassistant/components/xiaomi_miio/switch.py

```diff
@@ -113,120 +87,111 @@
 }
 
 
-async def async_setup_platform(hass, config, async_add_entities, discovery_info=None):
-    """Set up the switch from config."""
-    if DATA_KEY not in hass.data:
-        hass.data[DATA_KEY] = {}
-
-    host = config[CONF_HOST]
```

Comment:
Would it be possible to somehow import the old platform config to a config entry?

Reply:
In principle the required information is there.
But I don't know how to do that, Can I just call `async_step_user` directly with already supplying the user-input?

Reply:
Here's an example:
https://github.com/home-assistant/core/blob/2f40f44670b603caa63abe07e27126ca18a5aa3b/homeassistant/components/awair/sensor.py#L41-L52

Reply:
I added the yaml import to config flow, could you check if I did it alright?

I think I don't need to do any checks in the async_step_import since all nessesary checks are already in the async_step_device (like _abort_if_unique_id_configured())

Reply:
@rytilahti tested it for me and it works.